### DISCUSSION TITLE
Support for netrw / vim-vinegar style file browsing

### DISCUSSION THREAD

**Discussion by WesleyAC** [helix-editor/helix#9302] [Ideas]
Discussions about file browsing in Helix have focused on tree-style side panels. (See [here](https://github.com/helix-editor/helix/pull/5768#issuecomment-1722508888) for current status of that). However, I much prefer [netrw](https://vonheikemen.github.io/devlog/tools/using-netrw-vim-builtin-file-explorer/) / [vim-vinegar](https://github.com/tpope/vim-vinegar) style browsing, which is significantly simpler:

* From a file, press `-` to view a list of files that are in the same directory as the file you were editing. You can just `j`/`k` to move up and down, and press enter to open a file
* From a directory, press `-` again to move to the parent directory.

This allows for a few very nice things:

* Seeing what other files are in the current directory (simply press `-`, then press enter when you want to return to the file you were editing)
* Navigating to another file in the same directory or a nearby directory with very few keystrokes

I understand that some people prefer fuzzy-finding, and that can be better for some cases. However, I find this kind of quick UI very useful.

Helix already has something that seems at first glance to be similar to this — `file_picker_in_current_buffer_directory` — but that doesn't allow for navigating further up, nor using `j`/`k` to navigate the files.

Is this something that it would be reasonable to have in the core of Helix, or is this a "wait until there's a plugin system" situation?

**Comment by WesleyAC** [comment]
[This post](http://vimcasts.org/blog/2013/01/oil-and-vinegar-split-windows-and-project-drawer/) has much more detailed motivation for this paradigm, which may be interesting to people here.

**Comment by kirawi** [comment]
Ref https://matrix.to/#/!zMuVRxoqjyxyjSEBXc:matrix.org/$AbW0awK9XrJIHw6IqvcpXMxkhWt7apcgo-_8ndq5Qhs?via=matrix.org&via=mozilla.org&via=envs.net

**Reply by markstos** [reply]
Could you summarize what was interesting from that Matrix conversation here?
